ConsentViewController/Classes/SPSDK.swift

```diff
@@ -39,6 +39,7 @@ import Foundation
         legIntCategories: [String],
         handler: @escaping (SPGDPRConsent) -> Void
     )
+    @objc func logErrorMetrics(_ error: SPError)
```

Comment:
@Nevazhnovu I'm not sure we should expose `logErrorMetrics` method. This method seems very internal to the SDK in my opinion.

Reply:
@andresilveirah Yes it should be internal. But 1) using `onError` will trigger clearing all data from storage which is not desirable 2) we cannot declare `private` methods in protocols. What's the best option here?

Reply:
@andresilveirah I'm not used to make logs to `custom_metrics` so this entire commit may not be done in a smart way. Looking forward to see your suggestions!